src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -410,5 +446,56 @@ private IEnumerable<Snippet> GetRequiredPropertiesForObjectType(TypeSymbol typeS
                 }
             }
         }
+
+        public IEnumerable<Snippet> GetNestedResourceDeclarationSnippets(TypeSymbol typeSymbol)
+        {
+            // Leaving out the API version on this, because we expect its more common to inherit from the containing resource.
+            yield return new Snippet(@"resource ${1:Identifier} '${2:Type}' = {
+  name: $3
+  properties: {
+    $0
+  }
+}", prefix: "resource-with-defaults", detail: "Nested resource with defaults");
+
+            yield return new Snippet(@"resource ${1:Identifier} '${2:Type}' = {
+  name: $3
+  $0
+}", prefix: "resource-without-defaults", detail: "Nested resource without defaults");
+
+            string parentType = typeSymbol.Name;
+
+            if (resourceTypeToChildTypeSymbolsMap.ContainsKey(parentType))
+            {
+                foreach (TypeSymbol nestedTypeSymbol in resourceTypeToChildTypeSymbolsMap[typeSymbol.Name])
+                {
+                    resourceTypeInfoMap.TryGetValue(nestedTypeSymbol.Name, out (string prefix, string identifier, string bodyText, string description) resourceInfo);
+
+                    foreach (string nestedType in GetNestedResourceTypes(parentType, nestedTypeSymbol))
+                    {
+                        // The property "parent" is not allowed in nested resource. We'll remove the property before creating the snippet 
+                        string bodyText = ParentPropertyPattern.Replace(resourceInfo.bodyText, string.Empty);
```

Comment:
At some point, we should try to work towards storing snippets as Bicep syntax trees rather than strings. String find+replace will work most of the time, but e.g. if a snippet contains a legitimate property named "parent", or if we have snippets with parent properties spanning multiple lines in the future, this logic will break. I'll raise an issue for that.

Reply:
Raised #3384. I don't think you **need** to address this in this PR.

Reply:
Thanks Anthony for the suggestion and logging issue.

Reply:
Btw, until that's done, we should add a test that fails snippets that do contain strings that we replace to avoid weird bugs.

Reply:
I added couple of tests as part of this commit:
https://github.com/Azure/bicep/commits/dev/bhsubra/ShowChildSnippets

Let me know if I am missing something here.